### ISSUE TITLE
Textures missing after import

### ISSUE THREAD

**Issue by nwhite89** [soupday/cc_blender_tools#283]
When importing multiple FBX character file multiple textures seem to be lost. I can see visually on the initial import it is there and it seems to be lost after script starts to run. 

If I import the FBX file (through normal blender import) and change the mesh on the Character to that of the normal import textures all come back and seem to work but was hoping / wondering if there was a different way of fixing this or perhaps there was a known bug for this?

Thanks in advance

**Comment by soupday** [comment]
The problem is that when exporting multiple characters in one fbx from iClone, the exporter does not cope with multiple objects with the same name and with the name material names, and the textures overwrite each other.

So for CC3+ standard characters who all have CC_Base_Body/Std_Skin_* meshes and materials, they overwrite each other.

Some of the textures (Diffuse and Normal) are embedded in the Fbx and could be recovered, but the rest are lost.

Basically, don't export multiple CC3+ standard characters in one FBX, it's never worked properly.

**Comment by soupday** [comment]
Where are the characters from? This add-on is mostly just for importing from iClone and Character Creator.

It has a go at importing from other sources, but there's no guarantees it will work with them. Especially FBX files, they have such poor texture support, most of the textures just aren't there. It works quite well with GLTF / GLB's though.

But the majority of the functions in this add-on (Rigging / Spring bones / Sculpting) are exclusively for the CC3+ standard type and Rigging support for the ActorCore and AccuRig characters.

**Comment by nwhite89** [comment]
I appreciate that which is why I closed because it's not strictly a bug so really appreciate it @soupday. The files have all been created in Blender and have since got a Mixamo rig attached to them. I'm currently using this add on to convert the Mixamo rig into rigify rig which all works great I might add the only part is some of the textures get lost. If I change the mesh back to the original they all then work back to as expected but I am basically looking at doing about 10,000 models which is why I raised and asked this as I wasn't sure if it was like a known bug or something.

**Comment by nwhite89** [comment]
Appreciate all your help here @soupday I did some more investigation work here on bits and pieces and managed to see what's happening to the textures (no idea why mind) but thought would at least pass this information on to you.

So the original FBX uses some textures surface `Principled BSF` 

<img width="1128" alt="image" src="https://github.com/user-attachments/assets/4d037b3f-6955-47f3-a4fd-6d135be52355">

Now using the import via the add-on I actually notice that these textures change surface type to `Pbr Shader`.

<img width="1397" alt="image" src="https://github.com/user-attachments/assets/e0faebee-0f3d-499e-a9e6-ebab5768b246">

Now as a work around for the time being I'm planning on importing the FBX exporting a temp one to ensure all materials are 100% unique (as the number seems to change through the add-on as well) and then use this to remap back before completion. 

So I do have a work around no idea if you would want to class this as a bug or not (maybe? 🤷) but happy to share anything else based on your thoughts also.

Thanks again

**Comment by soupday** [comment]
Because it's for converting models to transfer between Blender and CC/iClone, when it imports any non CC models, it converts the materials into a custom shader that is directly compatible with the materials in CC/iClone. Which is not what you want.

It sounds to me that you don't actually need most of what this add-on actually does. You might be better off modifying the add-on's code or just extracting the bits you do actually use. 

How are you converting a mixamo rig into a rigify one? This add-on can't do that. At least not by itself.
